crates/fuel-core/src/service/adapters.rs

```diff
@@ -85,6 +86,155 @@ impl StaticGasPrice {
     }
 }
 
+#[cfg(test)]
+mod arc_gas_price_estimate_tests {
+    #![allow(non_snake_case)]
+
+    use super::*;
+    use proptest::proptest;
+
+    async fn _worst_case__correctly_calculates_value(
+        gas_price: u64,
+        starting_height: u32,
+        block_horizon: u32,
+        percentage: u16,
+    ) {
+        // given
+        let subject = ArcGasPriceEstimate::new(starting_height, gas_price, percentage);
+
+        // when
+        let target_height = starting_height.saturating_add(block_horizon);
+        let actual = subject
+            .worst_case_gas_price(target_height.into())
+            .await
+            .unwrap();
+
+        // then
+        let mut expected = gas_price;
+
+        for _ in 0..block_horizon {
+            let change_amount = expected
+                .saturating_mul(percentage as u64)
+                .saturating_div(100);
+            expected = expected.saturating_add(change_amount);
+        }
+
+        assert!(actual >= expected);
```

Comment:
I think we can be more precise here, and place an upper bound on actual, e.g.
```suggestion
        assert!(actual >= expected);
        assert!(actual <= expected.saturating_add(2000.0))
```

Reply:
Sure. But we don't actually care about the lower bound of this. In reality, the gas price given by the block producer will be much lower than what we estimate 99% of the time. So we only really care that the block producer never gives a value higher than what we estimate.